strawberry/field.py

```diff
@@ -127,8 +130,9 @@ def base_resolver(self, resolver: StrawberryResolver) -> None:
         self._base_resolver = resolver
         self.origin = resolver.wrapped_func
 
-        # Don't add field to __init__ or __repr__ once it has a resolver
+        # Don't add field to __init__, __repr__ and __eq__ once it has a resolver
         self.init = False
+        self.compare = False
```

Comment:
Should we not be comparing fields with resolvers? If I have a type with only fields with resolvers, should instances _always_ be equal?

Reply:
I'd say yes, how would you compare resolvers?

Reply:
Is executing them unreasonable? I guess only if they have no arguments

Reply:
@BryceBeagle I'll merge this for now, I think it makes sense to compare just values :)

Reply:
> Is executing them unreasonable? I guess only if they have no arguments

oh sorry, I missed your comment. I'd say so, also there might be side effects too :)